services/wallet/reader.go

```diff
@@ -461,84 +461,19 @@ func (r *Reader) createBalancePerChainPerSymbol(
 }
 
 func (r *Reader) GetWalletToken(ctx context.Context, clients map[uint64]chain.ClientInterface, addresses []common.Address, currency string) (map[common.Address][]token.StorageToken, error) {
-	cachedTokens, err := r.getCachedWalletTokensWithoutMarketData()
-	if err != nil {
-		return nil, err
-	}
-
-	chainIDs := maps.Keys(clients)
-
 	currencies := make([]string, 0)
 	currencies = append(currencies, currency)
 	currencies = append(currencies, getFixedCurrencies()...)
-	allTokens, err := r.tokenManager.GetTokensByChainIDs(chainIDs)
-	if err != nil {
-		return nil, err
-	}
 
-	tokenAddresses := getTokenAddresses(allTokens)
-
-	balances, err := r.tokenManager.GetBalancesByChain(ctx, clients, addresses, tokenAddresses)
+	result, err := r.FetchOrGetCachedWalletBalances(ctx, clients, addresses, true)
 	if err != nil {
-		log.Info("tokenManager.GetBalancesByChain error", "err", err)
 		return nil, err
 	}
 
-	verifiedTokens, unverifiedTokens := splitVerifiedTokens(allTokens)
 	tokenSymbols := make([]string, 0)
-	result := make(map[common.Address][]token.StorageToken)
-
-	for _, address := range addresses {
-		for _, tokenList := range [][]*token.Token{verifiedTokens, unverifiedTokens} {
-			for symbol, tokens := range getTokenBySymbols(tokenList) {
-				balancesPerChain := make(map[uint64]token.ChainBalance)
-				decimals := tokens[0].Decimals
-				isVisible := false
-				for _, tok := range tokens {
-					hexBalance := balances[tok.ChainID][address][tok.Address]
-					balance := big.NewFloat(0.0)
-					if hexBalance != nil {
-						balance = new(big.Float).Quo(
-							new(big.Float).SetInt(hexBalance.ToInt()),
-							big.NewFloat(math.Pow(10, float64(decimals))),
-						)
-					}
-					hasError := false
-					if client, ok := clients[tok.ChainID]; ok {
-						hasError = err != nil || !client.IsConnected()
-					}
-					if !isVisible {
-						isVisible = balance.Cmp(big.NewFloat(0.0)) > 0 || isCachedToken(cachedTokens, address, tok.Symbol, tok.ChainID)
-					}
-					balancesPerChain[tok.ChainID] = token.ChainBalance{
-						RawBalance: hexBalance.ToInt().String(),
-						Balance:    balance,
-						Address:    tok.Address,
-						ChainID:    tok.ChainID,
-						HasError:   hasError,
-					}
-				}
-
-				if !isVisible && !belongsToMandatoryTokens(symbol) {
-					continue
-				}
-
-				walletToken := token.StorageToken{
-					Token: token.Token{
-						Name:          tokens[0].Name,
-						Symbol:        symbol,
-						Decimals:      decimals,
-						PegSymbol:     token.GetTokenPegSymbol(symbol),
-						Verified:      tokens[0].Verified,
-						CommunityData: tokens[0].CommunityData,
-						Image:         tokens[0].Image,
-					},
-					BalancesPerChain: balancesPerChain,
-				}
-
-				tokenSymbols = append(tokenSymbols, symbol)
-				result[address] = append(result[address], walletToken)
-			}
+	for _, storageTokens := range result {
+		for _, t := range storageTokens {
+			tokenSymbols = append(tokenSymbols, t.Token.Symbol)
```

Comment:
Does this mean that we will just store all tokens regardless of whether they are verified or regardless of their visibility status?

Reply:
not sure actually :) I just return all the info

Reply:
That works 🙈